ui/app/AppLayouts/Wallet/controls/NetworkFilter.qml

```diff
@@ -94,23 +98,38 @@ Item {
         }
     }
 
-    NetworkSelectPopup {
-        id: selectPopup
-        x: (parent.width - width + 5)
-        y: (selectRectangleItem.height + 5)
-        layer1Networks: root.layer1Networks
-        layer2Networks: root.layer2Networks
-        testNetworks: root.testNetworks
-        multiSelection: root.multiSelection
-
-        onToggleNetwork: {
-            root.toggleNetwork(network.chainId)
-        }
+    Loader {
+        id: selectPopupLoader
+
+        active: false
+
+        sourceComponent: NetworkSelectPopup {
+            id: selectPopup
 
-        onSingleNetworkSelected: {
-            d.selectedChainName = chainName
-            d.selectedIconUrl = iconUrl
-            root.singleNetworkSelected(chainId, chainName, iconUrl)
+            readonly property point refPos: selectRectangleItem.mapToItem(parent, 0, 0)
+            x: Math.min(parent.width - width + 5, refPos.x)
+            y: Math.min(parent.height - height + 5, refPos.y + selectRectangleItem.height + 5)
+
+            layer1Networks: root.layer1Networks
+            layer2Networks: root.layer2Networks
+            testNetworks: root.testNetworks
+
+            multiSelection: root.multiSelection
+            useEnabledRole: false
+
+            onToggleNetwork: (network) => {
```

Comment:
When the `multiSelection/singleSelection` option was introduced, this component was copled with a specific `toggleNetwork` behavior in wallet so it was just introduced another signal `singleNetworkSelected` to avoid doing a big refactor on it. 
However, since now you are doing a big refactor here, would it make sense to just keep one signal, behavior agnostic, that is raised whenever the user selects/unselects an option by the UI (passing the needed info: `chainId`, `chainName` and `iconUrl`) and the consumer is in charge of doing whatever is needed, so we only have here a component with a "networkSelected/Unselected" signal for both cases (multi/single selection)?